### ISSUE TITLE
do not automatically add sparse: true to unique indexes

### ISSUE THREAD

**Issue by elennaro** [balderdashy/sails-mongo#360]
It's better to add it explicitly from the model:
````
attributes: {
		phone: {
			type: 'string',
			index: true,
			unique: true,
			// sparse: true, add support to this
			size: 20
		},
````

or at least add support to `sparse: false`.

Because of this: [Sparse Index On A Collection Cannot Return Complete Results](https://docs.mongodb.org/manual/core/index-sparse/#sparse-index-incomplete-results).

**Comment by Josebaseba** [comment]
Hi @elennaro, you can use this hook: https://github.com/Josebaseba/sails-hook-mongoat to set advanced options and multiple indexes to your mongo database. I'm using it in production and it increased radically my queries response time... 

Nowadays sails-mongo hook has some limitations (I think that's normal in an ORM) that we can implement via hooks.

Hope it helps!

**Comment by elennaro** [comment]
Great! Thank you @Josebaseba!

**Comment by sailsbot** [comment]
Thanks for posting, @elennaro.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
